mlir/python/mlir/_mlir_libs/_mlir/dialects/pdl.pyi

```diff
@@ -26,15 +25,15 @@ class AttributeType(Type):
   def isinstance(type: Type) -> bool: ...
 
   @staticmethod
-  def get(context: Optional[Context] = None) -> AttributeType: ...
+  def get(context: Context | None = None) -> AttributeType: ...
```

Comment:
this syntax requires 3.10? 3.11? ie would break 3.8, 3.9.

Reply:
It will break if the type checker enforces a specific Python version on type stubs. Not sure if that happens in practice.

Note also that this syntax is already used in [`ir.pyi`](https://github.com/llvm/llvm-project/blob/main/mlir/python/mlir/_mlir_libs/_mlir/ir.pyi).

Reply:
Is there a reason why MLIR wants to target 3.8+ for its Python bindings? 3.8 is EOL in a months and 3.9 is only receiving security updates before EOLing next year.

Reply:
Traditionally, llvm debates such things. I can't remember the specifics of the last time, but generally we've been tracking the eol schedule. Dropping 3.8 on those grounds SGTM.

Reply:
Type stubs are not used at all at runtime, so as long as type checkers are happy with the new syntax, it should be fine to use it, right?

Reply:
> It will break if the type checker enforces a specific Python version on type stubs.

That's not correct - it will raise a `TypeError` in just vanilla python:

```python
$ conda create -n throwaway python=3.8
$ conda activate throwaway
(throwaway) $ python -c "def fun(a: int | float): pass"
Traceback (most recent call last):
  File "<string>", line 1, in <module>
TypeError: unsupported operand type(s) for |: 'type' and 'type'
```

Reply:
I don't have a strong opinion on moving the support window forward - I'm just verifying.

Reply:
Ah sorry I didn't realize this was in pyi file. In which case I think you're correct.